core/src/main/java/org/apache/iceberg/RemoveSnapshots.java

```diff
@@ -163,21 +180,86 @@ private TableMetadata internalApply() {
     this.base = ops.refresh();
 
     Set<Long> idsToRetain = Sets.newHashSet();
-    List<Long> ancestorIds = SnapshotUtil.ancestorIds(base.currentSnapshot(), base::snapshot);
-    if (minNumSnapshots >= ancestorIds.size()) {
+    // Identify refs that should be removed
+    Set<String> refsToRemove = Sets.newHashSet();
+    for (String name : base.refs().keySet()) {
+      SnapshotRef ref = base.ref(name);
+      if (!name.equals(SnapshotRef.MAIN_BRANCH) && ref.maxRefAgeMs() != null) {
```

Comment:
I don't think this is quite right. The max ref age table setting should be applied if the ref itself has no maximum age. Instead, this just skips the ref.

Reply:
You are correct, I missed in the spec, there's a table level property for max-ref-age-ms that we want to default to. The same applies for the other retention properties. This behavior makes more sense. I will update to fix this.